### ISSUE TITLE
Trailing commas

### ISSUE THREAD

**Issue by johanneskoester** [snakemake/snakefmt#24]
Black on Python adds a trailing comma to multiline argument lists. While this is good practice for programming, the situation is different for workflow definitions. Here, maximum readability is more important than a (arguably only a little) cleaner git history. Hence, snakefmt should ideally not enforce and even remove trailing commas in all cases. Can black be configured like that? If so, this should be default for snakefmt.

**Comment by bricoletc** [comment]
Hi @johanneskoester , do you mean removing trailing commas from python code, or removing them from things like

```
rule a:
    input:
        "a",
        "b",
```

If the latter, this is not produced by black but by custom code and can easily be removed

The reason it is there is not just for git history but also @mbhall88 saying that he often finds adding an argument causes less errors, you just copy paste it and don't need to add a comma. Thoughts?

**Comment by johanneskoester** [comment]
True as well, but readability and clean code is the highest goal for a workflow. More important than avoiding developer errors. Hence, yes, I strongly vote for omitting trailing commas.

Doing it for the snakemake part for now seems enough to me for now.

**Comment by johanneskoester** [comment]
Mhm, further thinking about it, it is also not clean if the rule is applied inconsistently, comma for python, no comma for snakemake. Maybe it is better to add it everywhere if black cannot be convinced to not do it in python code.

**Comment by bricoletc** [comment]
Indeed, by design black is not configurable: the function we [call](https://black.readthedocs.io/en/stable/reference/reference_functions.html#black.format_str) only accepts configuration for line length and string normalisation (which quotes to prefer)

**Comment by johanneskoester** [comment]
Ok, trailing commas everywhere then.

**Comment by jakevc** [comment]
We re-discussed this issue and are potentially favoring no trailing commas for single parameters again
